cdap-examples/Purchase/src/test/java/co/cask/cdap/examples/purchase/PurchaseAppTest.java

```diff
@@ -142,6 +145,18 @@ private ServiceManager getServiceManager(ApplicationManager appManager) throws I
     return userProfileServiceManager;
   }
 
+  private String getServiceUrlWithRetry(ServiceManager serviceManager) {
+    int trial = 0;
+    while (trial++ < 5) {
+      URL url = serviceManager.getServiceURL(15, TimeUnit.SECONDS);
+      if (url != null) {
```

Comment:
Ok my bad. I thought that if it were null, the malformed URL would be null/history/joe after concat. But just tested that ``new URL(null, "history/joe")`` throws a ``MalformedUrlException: history/joe``. So ignore my comment.

LGTM after the compilation error fix.